src/prism.c

```diff
@@ -6075,6 +6075,7 @@ parser_lex_magic_comment_encoding_value(pm_parser_t *parser, const uint8_t *star
                 ENCODING1("CP860", pm_encoding_ibm860);
                 ENCODING1("CP861", pm_encoding_ibm861);
                 ENCODING1("CP862", pm_encoding_ibm862);
+                ENCODING1("CP866", pm_encoding_ibm866);
```

Comment:
Wasn't sure if this should be added because of [this](https://github.com/ruby/prism/pull/1853/files#r1397439189) but `CP866` tests fail if I don't.

Reply:
This is correct! We need to check the string to ensure

```ruby
# encoding: cp866
```

changes the encoding properly.